### PR TITLE
provider/lxd: cache certs in juju/lxd dir

### PR DESCRIPTION
**Pull request by axw**
## Description of change

When bootstrapping LXD, we will now cache the generated credential in ~/.local/share/juju/lxd. This allows us to use the same certificates each time we bootstrap, rather than a fresh one each time. Since we'll only generate one, and upload it once, we no longer need to remove certs on controller destruction.

## QA steps

juju bootstrap localhost b
juju bootstrap localhost a
juju add-model foo
juju migrate foo b
(confirm model migrated)
juju destroy-controller -y a
juju destroy-controller -y b

juju bootstrap localhost (with juju 2.0)
juju destroy-controller -y localhost-localhost (with this branch)

juju bootstrap localhost (with juju 2.0)
juju upgrade-juju -m controller (with this branch)
juju destroy-controller -y localhost-localhost (with this branch)

## Documentation changes

No change.

## Bug reference

Fixes https://bugs.launchpad.net/juju/+bug/1661091

### CODE REVIEW

File: provider/lxd/credentials.go [hunk 1]
```diff
@@ -52,69 +57,172 @@ func (environProviderCredentials) CredentialSchemas() map[cloud.AuthType]cloud.C
...
+	certPEM, keyPEM, err := readCert(jujuLXDDir)
+	if err == nil {
+		return certPEM, keyPEM, nil
+	} else if !os.IsNotExist(err) {
+		return nil, nil, errors.Trace(err)
+	}
+
+	// Next we look in the LXD config dir, in case the user has
+	// a client certificate/key pair for use with the "lxc" client
+	// application.A
```
Comment (wallyworld) [inline provider/lxd/credentials.go]: typo

Comment (axw) [inline provider/lxd/credentials.go]: fixed

**Comment by axw** [comment]
> All that nice new lxd provider code from a day or so ago gone.

I like to think that it was replaced with nicer code at least ;)

**Comment by axw** [comment]
$$__fixes-1661091__$$

**Comment by axw** [comment]
$$howaboutnow$$

**Comment by wallyworld** [comment]
$$__JFDI__$$

**Comment by axw** [comment]
$$__JFDI__$$

**Pull request merged**